components/react/hooks/src/useDynamicHeightCollapsible/index.js

```diff
@@ -0,0 +1,17 @@
+import {useState, useCallback} from 'react'
+import useToggle from '../useToggle'
+
+export default function useDynamicHeightCollapsible(getContentHeighFn) {
+  const [isCollapsed, toggleCollapsible] = useToggle(true)
+  const [contentHeight, setContentHeight] = useState(null)
+  const heightRecalculator = useCallback(() => {
+    const newContentHeight = getContentHeighFn()
+    setContentHeight(newContentHeight)
+  }, [getContentHeighFn])
+
+  return [
+    isCollapsed,
+    contentHeight,
+    {toggle: () => toggleCollapsible(), heightRecalculator}
```

Comment:
`toggle` will always be a new function if declared like this, use the already memorized function from useToggle:
```jsx
const [isCollapsed, toggle] = useToggle(true)
...
{toggle, heightRecalculator}
```

Reply:
Yeah, but I need the toggle to not allow passing an initialValue since the SUI-MoleculeCollapsible does not allow to specify an initial value for the open/close state

Reply:
Mmm what initial value? the toggle function accepts only  a boolean as argument, otherwise it'll be ignored. It should be safe if this was the problem, otherwise I need to better understand the specific case you are talking about 😃

Reply:
done!